.github/workflows/ci.yml

```diff
@@ -0,0 +1,271 @@
+          make package BUILD_NUMBER='${GITHUB_HEAD_REF}-${GITHUB_RUN_ID}'
+      - name: Persist dist artifacts
+        uses: actions/upload-artifact@0b7f8abb1508181956e8e162db84b466c27e18ce # v3.1.2
+        with:
+          name: server-dist-artifact
+          path: dist/
+          retention-days: 14
+      - name: Persist build artifacts
+        uses: actions/upload-artifact@0b7f8abb1508181956e8e162db84b466c27e18ce # v3.1.2
+        with:
+          name: server-build-artifact
+          path: build/
+          retention-days: 14 
+  upload-s3:
+    name: Upload to S3 bucket
+    runs-on: ubuntu-latest-8-cores
+    needs: 
+      - build-mattermost-server
+      - test-mysql
+      - test-postgres-binary
+      - test-postgres-normal 
+    env:
+      REPO_NAME: ${{ github.event.repository.name }}
+    steps:
+    - name: Download dist artifacts
+      uses: actions/download-artifact@e9ef242655d12993efdcda9058dee2db83a2cb9b  # v3.0.2
+      with:
+        name: server-dist-artifact
+        path: dist/
+    - name: Configure AWS
+      uses: aws-actions/configure-aws-credentials@07c2f971bac433df982ccc261983ae443861db49 # v1-node16
+      with:
+        aws-region: us-east-1
+        aws-access-key-id: ${{ secrets.MM_SERVER_AWS_ACCESS_KEY_ID }}
+        aws-secret-access-key: ${{ secrets.MM_SERVER_AWS_SECRET_ACCESS_KEY }}
+    # We need to sanitize the branch name before using it
+    - name: ci/sanitize-branch-name
+      id: branch
+      uses: transferwise/sanitize-branch-name@b10b4d524ac5a7b645b43a3527db3a6cca017b9d # v1
+    # Search for the string "pull" and replace it with "PR" in branch-name
+    - name: ci/sanitize-branch-name-replace-pull-with-PR-
+      run: echo "BRANCH_NAME=$(echo ${{ steps.branch.outputs.sanitized-branch-name }} | sed 's/^pull\//PR-/g')" >> $GITHUB_ENV
+    - name: ci/artifact-upload
+      run: |
+        aws s3 cp dist/ s3://pr-builds.mattermost.com/$REPO_NAME/$BRANCH_NAME/ --acl public-read --cache-control "no-cache" --recursive
+        aws s3 cp dist/ s3://pr-builds.mattermost.com/$REPO_NAME/commit/${{ github.sha }}/ --acl public-read --cache-control "no-cache" --recursive
+  build-docker:
+    name: Build docker image
+    runs-on: ubuntu-latest-8-cores
+    needs: upload-s3
+    steps:
+    - name: Download build artifacts
+      uses: actions/download-artifact@e9ef242655d12993efdcda9058dee2db83a2cb9b  # v3.0.2
+      with:
+        name: server-build-artifact
+    - name: Login to Docker Hub
+      uses: docker/login-action@3da7dc6e2b31f99ef2cb9fb4c50fb0971e0d0139 # v2.1.0
+      with:
+        username: ${{ secrets.DOCKERHUB_USERNAME }}
+        password: ${{ secrets.DOCKERHUB_TOKEN }}
+    - name: Setup Docker Buildx
+      uses: docker/setup-buildx-action@11e8a2e2910826a92412015c515187a2d6750279 # v2.4
+    - name: Docker build and push
+      env:
+        DOCKER_CLI_EXPERIMENTAL: enabled
+      run: |
+        export TAG="${GITHUB_SHA:0:7}"
```

Comment:
Is this correct? 

We've received a PR in webapp repo, from @saturninoabril  to change this. 
https://github.com/mattermost/mattermost-webapp/pull/12214

Reply:
It seems to be correct per the circleci config, we can opt to use github variables if that one is more reliable/consistent/convenient. I'm 0/5 on this.

Reply:
Since you are not triggering this workflow on `pull_request` you should be fine